### DISCUSSION TITLE
single-commit: true doesn't remove old deployments?

### DISCUSSION THREAD

**Discussion by chriscarrollsmith** [JamesIves/github-pages-deploy-action#1446] [Q&A] [answered]
The README suggests that setting `single-commit: true` will remove all old deployments, but I did not find that to be true after this workflow run:

https://github.com/babafaqirchand/babafaqirchand.github.io/actions/runs/6332179166/job/17198163171

I used:

```      - name: Deploy 🚀
        uses: JamesIves/github-pages-deploy-action@v4
        with:
          folder: dist
          branch: gh-pages
          single-commit: true
```

When I go to the gh-pages branch and view, deployments, I still see over 1,000 deployments:

https://github.com/babafaqirchand/babafaqirchand.github.io/deployments

Am I misunderstanding the usage or purpose of this option?

**Comment by JamesIves** [comment, marked as answer]
This option removes the **commit history** from the branch you are pushing code to. It cannot remove the deployment history from the GitHub Pages workflow as this isn't managed by the action, instead that's managed by an internal process from GitHub themselves. 

If you look at the commit history you'll see you have just a single commit on the `gh-pages` branch: https://github.com/babafaqirchand/babafaqirchand.github.io/commits/gh-pages/

```
This option can be toggled to true if you'd prefer to have a single commit on the deployment branch instead of maintaining the full history. Using this option will also cause any existing history to be wiped from the deployment branch.
```

**Comment by chriscarrollsmith** [comment]
Gotcha. Thanks for the clarification!
